### PR TITLE
fixed sidebar position, added toHide to <LicenseRequired>

### LINKED ISSUE #6227: [CAL-659] fix sidebar tip position
<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-659](https://linear.app/calcom/issue/CAL-659/fix-sidebar-tip-position)</sub>

### PR DESCRIPTION
**Pull request by PeerRich**
supersedes #6219 cc @G3root 

- fixed positioning of sidebar tips
- removed @JeroenReumkens's transparent fixed gradient due to issues 
- added `<LicenseRequired toHide>` to only hide it for people with a license (it'll show tips for non-EE self-hosters)

before:
CleanShot 2023-01-01 at 12 27 47

after:

CleanShot 2023-01-01 at 12 26 24

### CODE REVIEW

File: packages/features/ee/common/components/v2/LicenseRequired.tsx [hunk 1]
```diff
@@ -11,21 +11,24 @@ type LicenseRequiredProps = {
...
   const session = useSession();
   const { t } = useLocale();
   const Component = as || Fragment;
   const hasValidLicense = session.data ? session.data.hasValidLicense : null;
+
   return (
     <Component {...rest}>
       {hasValidLicense === null || hasValidLicense ? (
-        children
+        !toHide && children
```
Comment (PeerRich) [inline packages/features/ee/common/components/v2/LicenseRequired.tsx:31]: @zomars does this logic check out?

I want to:

hide the children if a license is valid

File: packages/ui/v2/core/Shell.tsx [hunk 2]
```diff
@@ -780,9 +781,10 @@ function SideBar() {
           <Navigation />
         </div>
 
-        {isCalcom && <Tips />}
-
-        <div className="fixed left-1 bottom-0 w-4 bg-gray-50 before:absolute before:left-0 before:-top-20 before:h-20 before:w-48 before:bg-gradient-to-t before:from-gray-50 before:to-transparent lg:left-4 lg:w-48">
+        <div>
+          <LicenseRequired toHide>
+            <Tips />
+          </LicenseRequired>
```
Comment (PeerRich) [inline packages/ui/v2/core/Shell.tsx:787]: previously we've shown the tips only on cal.com, i think its a good idea to show the tips for CE (non-EE) self-hosters, too

File: packages/ui/v2/core/Shell.tsx [hunk 3]
```diff
@@ -780,9 +781,10 @@ function SideBar() {
           <Navigation />
         </div>
 
-        {isCalcom && <Tips />}
-
-        <div className="fixed left-1 bottom-0 w-4 bg-gray-50 before:absolute before:left-0 before:-top-20 before:h-20 before:w-48 before:bg-gradient-to-t before:from-gray-50 before:to-transparent lg:left-4 lg:w-48">
```
Comment (PeerRich) [inline packages/ui/v2/core/Shell.tsx:785]: @JeroenReumkens this was a neat idea but caused a lot of headaches with the `<Tips>` component

File: packages/ui/v2/core/Shell.tsx [hunk 4]
```diff
@@ -780,9 +781,10 @@ function SideBar() {
...
+        <div>
+          <LicenseRequired toHide>
```
Comment (JeroenReumkens) [inline packages/ui/v2/core/Shell.tsx:785]: This now shows a enterprise error when viewed on a platform without enterprise key. That's not desired I think? 

CleanShot 2023-01-02 at 09 30 01@2x

File: packages/ui/v2/core/Shell.tsx (hunk 3 above)
Comment (JeroenReumkens) [reply packages/ui/v2/core/Shell.tsx:785]: Ah damn... Totally forgot about that component 😢 Perhaps the tips could be nested inside the fixed wrapper. Want me to try this still?

**Review by JeroenReumkens — DISMISSED** [review]
One small change, apart from that good change!

File: packages/ui/v2/core/Shell.tsx (hunk 4 above)
Comment (PeerRich) [reply packages/ui/v2/core/Shell.tsx:785]: oh right good point. it should show the children without License. (if toHide is true)

Comment (PeerRich) [reply packages/ui/v2/core/Shell.tsx:785]: can you check if this was resolved? with my latest commit

Comment (PeerRich) [reply packages/ui/v2/core/Shell.tsx:785]: https://github.com/calcom/cal.com/pull/6228/commits/f65422b0e188202ecb6d9bdc9ffbe00193803505

**Comment by JeroenReumkens** [comment]
@PeerRich  Yes amazing, this works!

**Review by JeroenReumkens — APPROVED** [review]

**Pull request merged**